editor/components/post-type-support-check/index.js

```diff
@@ -1,29 +1,37 @@
 /**
  * External dependencies
  */
-import { get, some, castArray } from 'lodash';
+import { get, includes, some, castArray } from 'lodash';
 
 /**
  * WordPress dependencies
  */
 import { withSelect } from '@wordpress/data';
 
-function PostTypeSupportCheck( { postType, children, supportKeys } ) {
+export function PostTypeSupportCheck( { postType, children, supportKeys, themeSupports } ) {
+	supportKeys = castArray( supportKeys );
 	const isSupported = some(
-		castArray( supportKeys ), ( key ) => get( postType, [ 'supports', key ], false )
+		supportKeys, ( key ) => get( postType, [ 'supports', key ], false )
 	);
 
 	if ( ! isSupported ) {
 		return null;
 	}
 
+	// 'thumbnail' and 'post-thumbnails' are intentionally different.
+	if ( includes( supportKeys, 'thumbnail' ) &&
+		! get( themeSupports, 'post-thumbnails', false ) ) {
```

Comment:
This must be updated to pass the second argument as an array, or will fail master build. (#6247)